torch/distributed/tensor/_random.py

```diff
@@ -71,20 +71,13 @@ def manual_seed(seed: int, device_mesh: DeviceMesh) -> None:
             f"DTensor randomness only supports cuda/cuda-like device type, but got {device_mesh.device_type}"
         )
 
-    # allgather the seed over the default PG
-    object_list = [seed] * dist.get_world_size()
-    dist.all_gather_object(object_list, seed)
-    for rank, object in enumerate(object_list):
-        if seed != int(object):
-            raise RuntimeError(
-                f"calling manual_seed function over {device_mesh} but received different seed values on ranks:",
-                f"seed on rank {dist.get_rank()} is {seed}, and seed on rank {rank} is {object}!",
-            )
     # instantiate a RNG tracker if haven't. By default DTensor uses an
     # OffsetBasedRNGTracker to perform random operators.
     global _rng_tracker
     if not _rng_tracker:
-        _rng_tracker = OffsetBasedRNGTracker(device_mesh.device_type)
+        _rng_tracker = OffsetBasedRNGTracker(
```

Comment:
what if a user calls manual seed a second time?  it looks like that would be a silent no-op, which is not expected.  should we remove the `if not _rng_tracker` check and just keep re-initializing new trackers each time manual_seed is called?  Or if there is some reason reinit is not supported, we should probably raise a RuntimeError here.

Reply:
not really. This if-branch only skips the RNG tracker initialization part. The seed setting remains being executed.